feat: UIG-2509 - vl-map-layer-switcher - dynamic layers

ksvlaanderen opened the pull request:
Refactor naar LitElement
Slot omgezet naar layers-property (breaking change)
Storybook uitgebreid
Cypress uitgebreid
WC testen verwijderd

Goldflow commented on apps/storybook-e2e/src/e2e/map/layer-switcher/vl-map-layer-switcher.stories-wc.cy.ts:
```diff
@@ -21,80 +21,49 @@ const shouldHaveInvisibleLayerFor = (layerName: string): void => {
+        cy.visit(`${mapLayerSwitcherDynamic}`);
 
+        shouldHaveFeatureLayerCount(0);
+        cy.get(`button#add-${layerId}`).click();
```
beter 👍🏼

Goldflow commented on libs/map/src/lib/components/layer-switcher/vl-map-layer-switcher.ts:
```diff
@@ -1,134 +1,144 @@
+                mutation.removedNodes.forEach((node) => {
+                    if ((node as HTMLElement).hasAttribute('data-vl-is-layer')) {
+                        this.vlMapLayers = this.vlMapLayers.filter(
+                            (layer) => (node as unknown as VlMapLayer).title !== layer.title
```
als afnemer voor bepaalde reden geen `data-vl-name` instelt, wordt default `kaartlaag` genomen als title

als dit voor meerdere lagen gebeurt dan worden die allemaal tegelijk verwijderd, dus dit is niet de meest veilige manier om elke laag op een unieke manier te identificeren.

Een manier waarop je het zou kunnen doen:

OpenLayers houd intern ook unique id's bij voor elke laag; ik stel voor om dan bv. op volgende manier de juiste lagen er uit te halen (tegengekomen bij verdubbeling van map-layer-styles)

`(layer) => getUid(node as unknown as VlMapLayer) !== getUid(layer)`

dan moet je bij imports nog dit toevoegen: 
`import { getUid } from 'ol/util';`

ksvlaanderen replied:
Very good spot thanks! :D

Ik heb het op een licht andere manier gefixt, ik vergelijk nu gewoon de layers als object met elkaar aangezien deze dezelfde referentie hebben.

Goldflow reviewed (changes requested):
refactor naar Lit ziet er heel goed uit (ook al heb ik nog wat uitleg nodig om alles te vatten)

1 opmerking ivm het dynamisch verwijderen van lagen

Goldflow commented on libs/map/src/lib/components/layer-switcher/vl-map-layer-switcher.ts:
```diff
@@ -1,134 +1,144 @@
+                mutation.removedNodes.forEach((node) => {
+                    if ((node as HTMLElement).hasAttribute('data-vl-is-layer')) {
+                        this.vlMapLayers = this.vlMapLayers.filter(
+                            (layer) => (node as unknown as VlMapLayer) !== layer
```
👍🏼

ksvlaanderen committed:
feat: UIG-2509 - vl-map-layer-switcher - dynamic layers

Refactor naar LitElement
Slot omgezet naar layers-property (breaking change)
Storybook uitgebreid
Cypress uitgebreid
WCTest verwijderd

kspeltix commented:
:tada: This PR is included in version 1.10.0 :tada:

The release is available on [GitHub release](https://github.com/milieuinfo/uigov-web-components/releases/tag/v1.10.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
